### PR TITLE
Using the /search endpoint

### PR DESCRIPTION
**Pull request by pmarjan**
Using the /search endpoint

whatsnew
Added the [Search for products with the /search endpoint](https://devdocs.magento.com/guides/v2.2/rest/search-endpoint.html) topic.

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
@pmarjan thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by dobooth** [comment]
Hi @pmarjan Thanks for your contribution! The red 'x's above note a couple issues: First, please sign the new Adobe CLA so we can get this merged. Second, you have some markdown errors. Click the Details link above to see them. When using unordered lists, there should be two spaces between the * and the text. This is so it aligns the same way as ordered lists. And if each version is identical, consider [symlinking](https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md#symbolic-links) them.

File: guides/v2.2/rest/search-endpoint.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+group: rest-api
+title: Search using the /search endpoint
+---
+
```
[the remark this answers is not in the document]
Comment (nuzil) [reply guides/v2.2/rest/search-endpoint.md]: @pmarjan can you please use following:

contributor_name: comwrap GmbH
contributor_link: https://www.comwrap.com/

File: guides/v2.2/rest/search-endpoint.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+---
+group: rest-api
+title: Search with the /search endpoint
+contributor_name: 
+contributor_link: 
+---
+
+By using `GET {{URL}}rest/V1/search`, you can search through the catalog. The obtained results will come from your configured search engine (**Stores** > Settings > **Configuration** > **Catalog** > **Catalog Search** > **Search engine**).
```
Comment (nuzil) [inline guides/v2.2/rest/search-endpoint.md]: "Settings" also as **Settings** i expect?

File: guides/v2.2/rest/search-endpoint.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+This means you can collect results from Elasticsearch, for example, that are very similar to the results that are visible by using the regular frontend search.
```
Comment (nuzil) [inline guides/v2.2/rest/search-endpoint.md]: I would not be strict here to which engine is used even as example, maybe better to write
"This means you can collect results from search engine that is enabled. And those results will be equal to the results that are coming by using the regular frontend search."

**Comment by nuzil** [comment]
Sorry, I closed and re-open PR cause I added Marjan to comwrap team and wanted that system recognised that

File: guides/v2.2/rest/search-endpoint.md (hunk 2 above)
Comment (keharper) [reply guides/v2.2/rest/search-endpoint.md]: Actually, no. You don't click on that

Comment (nuzil) [reply guides/v2.2/rest/search-endpoint.md]: ah got it, right

**Comment by pmarjan** [comment]
@keharper , @nuzil 
the requested changes are done.
Please let me know if anything extra is needed.

**Comment by keharper** [comment]
I don't know how to push filesystem changes, so I have to just give instructions on setting up a symlink from 2.2 to 2.3.

1. `cd <devdocs_root>/guides/v2.3/rest/`
2. `rm search-endpoint.md`
3. `ln -s ../../v2.2/rest/search-endpoint.md .`

I've been testing your examples and have some further comments. I'll try to add them by Monday.

**Comment by keharper** [comment]
Also, this PR should probably be rebased to `master`. I don't think anything discussed here is new technology--just undocumented. If so, this topic should be made available immediately, instead of waiting till the 2.3.4 release date.

**Comment by pmarjan** [comment]
@keharper 
the symlink is updated and commits are rabased to master.

**Comment by keharper** [comment]
This topic needs some information about when you'd use `GET V1/search` vs `GET V1/products`.  Some differences I see between the two:

* `search` can be run on behalf of a guest user.  `products` requires an admin token
* The syntaxes are similar but different. `search` requires one of 3 `requestName`s. I'm not sure if `products` is more flexible on the attributes it can search by default.
* `search` can return aggregation/bucket data

Provide descriptions of the 3 search types. `quick` and `advanced` is self-explanatory, but I have no idea what `catalog_view` really means.

File: guides/v2.2/rest/search-endpoint.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+*  quick_search_container
+*  catalog_view_container
+*  advanced_search_container
+
+All of them run against the `catalogsearch_fulltext` index.
+
+The basic usage is:
+
+```http
+GET <host>/rest/<store_code>/V1/search?searchCriteria[requestName]=quick_search_container&
```
Comment (keharper) [inline guides/v2.2/rest/search-endpoint.md]: I tried this query, except I used `name` as the search term.  The `name` attribute is set to **Use in Search** is set to Yes. Based on the statement in line 8, I would expect the search to work. `catalog_view_container` doesn't work either, but `advanced_search_container` does.

Comment (pmarjan) [reply guides/v2.2/rest/search-endpoint.md]: I think it is like this:
The `field` can be set to be `search_term` only in the `quick_search_container`. This means both `catalog_view` and `advanced` will not work in that case.
Please send the queries you use? I could check them.

**Comment by pmarjan** [comment]
I have now made new version where there is an attempt to outline some differences between the search types, but also between using the v1/search and v1/products. At the end small summary is added too.

**Comment by keharper** [comment]
@pmarjan These are some nice additions. I'm going to move things around a bit, but hopefully we can get this to its final state by tomorrow.

**Comment by pmarjan** [comment]
@keharper 
I saw your last additions. Much nicer to read now :)
But, is there anything else that I could/should do?

**Comment by keharper** [comment]
@pmarjan I forgot to tag you and ask for you to review my changes. Since you've done that, you're done. Thanks!

**Comment by dobooth** [comment]
running tests

**Comment by dobooth** [comment]
running tests

**Comment by dobooth** [comment]
running tests

**Pull request merged**